scripts/13-why-rust.md

```diff
@@ -0,0 +1,469 @@
+%%
+<style>
+:root {--r-code-font: "FiraCode Nerd Font";}
+</style>
+
+# Cargo.toml
+
+```toml
+[package]
+name = "template"
+version = "0.1.0"
+edition = "2021"
+
+[build-dependencies]
+
+[dev-dependencies]
+
+[dependencies]
+```
+
+# Lint Tweaks
+
+```rust
+#![allow(dead_code)]
+#![allow(unused_variables)]
+```
+
+# Extern Crates
+
+```rust
+
+```
+
+# Imports
+
+```rust
+use std::collections::BTreeMap;
+```
+
+# Setup
+
+```rust
+fn main() {
+	type Dependency = String;
+	println!("Rust is boring");
+
+```
+
+%%
+
+![[rust-logo.png]]
+
+# Rust is Boring
+
+### Why Rust?
+
+notes:
+
+Hi friends my name is Tris and this is No Boilerplate, focusing on fast, technical videos.
+
+Today I hope to show you that Rust is ready for use in your company, in production, NOW.
+
+Not only that, but if you're not using Rust, you're missing out on a sea-change in how we build software.
+
+---
+
+### Crate Definition From crates.io
+
+```rust
+struct Crate {
+    name: String,
+    vers: String,
+    deps: Vec<Dependency>,
+    cksum: String,
+    features: BTreeMap<String, Vec<String>>,
+    features2: Option<BTreeMap<String, Vec<String>>>,
+    yanked: Option<bool>,
+    links: Option<String>,
+    v: Option<u32>
+}
+```
+
+notes:
+
+Rust's package registry is called Crates.io.
+
+When running a package registry you have the problem 
+
+
+---
+
+# Some Stats
+
+- Rust can legally drive a car
+- crates.io has ~100k packages
+- First-class support on AWS, Azure, GCP
+- Most-used wasm language
+- Most-loved language since 1.0 (7 years)
+
+(stackoverflow developer survey)
+
+notes:
+
+Rust is 16 years old now, has more packages than npm did in 2012, has first-class support on all enterprise clouds, and in-browser with wasm for 5 years.
+
+It has been the most loved language according to the largest developer survey on the planet for 7 years, that's every single year since 1.0.
+
+---
+
+![[jack-rust-perfect.png]]
+
+<https://twitter.com/jack/status/1474263588651126788>
+
+notes:
+
+Why does everyone love Rust?
+
+This is Jack Dorsey, who built twitter.
+
+---
+
+![[failwhale.png]]
+
+notes:
+
+Twitter started out using Ruby on Rails, then switched to a scala backend and microservices.
+
+Jack didn't build twitter with C or C++. He's a web developer.
+
+---
+
+> If you almost never add anything, you get C.
+> If you almost never remove anything, you get C++.
+
+&mdash;Raffi Molero
+
+notes:
+
+I don't quite agree that rust is a PERFECT language, there are many features that I want in Rust, and every 6 weeks I get them, thanks to the incredible core team.
+
+I don't think Rust is perfect, I think that Rust can allow you to write perfect code.
+
+---
+
+<split even>
+
+<ul>
+<li>Correctness focus</li>
+<li>Algebraic data types</li>
+<li>Compile-time execution</li>
+<li>Built-in transpiler metaprogramming</li>
```

Comment:
Why "transpiler" metaprogramming specifically? Isn't it just metaprogramming?

Reply:
It's to be VERY clear to people who don't understand what macros are: Python and Ruby have metaprogramming, but not 'transpiler' - I'm hoping people understand that this means arbitrary syntax transformation

Reply:
🤔 fair